python-server/pvCamera.py

```diff
@@ -0,0 +1,232 @@
+        'y': settingSignals['sizeY'].get() - settingSignals['startY'].get(),
+        'colorMode': colorModeEnumList[colorModeValue],
+        'dataType': dataTypeEnumList[dataTypeValue]
+    }
+    buffer.put_nowait((None, time.time(), tempDimensions))
+
+# Main loop for streaming images
+
+async def handle_streaming(websocket, buffer):
+    try:
+        while True:
+            rawImageArray, timestamp, updatedSettings = await buffer.get()
+
+            if updatedSettings:
+                currentSettings = updatedSettings
+                await websocket.send_text(json.dumps(currentSettings))
+                continue
+
+            height, width, colorMode, dataType = currentSettings['y'], currentSettings['x'], currentSettings['colorMode'], currentSettings['dataType']
+
+            try:
+                array_data = np.array(rawImageArray, dtype=dtype_map[dataType])
+                array_data = normalize_array_data(array_data, dataType)
+                array_data, mode = reshape_array(array_data, height, width, colorMode)
+
+                await send_image(array_data, websocket, mode)
+            except Exception as e:
+                print(f'Skipping image due to error: {e}')
+                continue
+    except WebSocketDisconnect:
+        await websocket.close()
+
+def normalize_array_data(array_data, dataType):
+    if dataType not in ['UInt8', 'Int8']:
+        max_val = array_data.max() if array_data.max() > 0 else 1
+        array_data = (array_data / max_val * 255).astype(np.uint8)
+    return array_data
+
+def reshape_array(array_data, height, width, colorMode):
+    if colorMode == 'Mono':
+        reshaped_data = array_data.reshape((height, width))
+        mode = 'L'  # Grayscale
+    elif colorMode == 'RGB1':
+        reshaped_data = array_data.reshape((height, width, 3))
+        mode = 'RGB'
+    elif colorMode == 'RGB2':
+        # Reshape to (height, width * 3) and split each row into R, G, B channels
+        array_data = array_data.reshape((height, width * 3))
+        red = array_data[:, 0:width]
+        green = array_data[:, width:2*width]
+        blue = array_data[:, 2*width:3*width]
+        reshaped_data = np.stack((red, green, blue), axis=-1)
+        mode = 'RGB'
+    elif colorMode == 'RGB3':
+        red = array_data[0:height * width].reshape((height, width))
+        green = array_data[height * width:2 * height * width].reshape((height, width))
+        blue = array_data[2 * height * width:3 * height * width].reshape((height, width))
+        reshaped_data = np.stack((red, green, blue), axis=-1)
+        mode = 'RGB'
+    else:
+        raise ValueError(f"Unsupported color mode: {colorMode}")
+    
+    return reshaped_data, mode
+
+async def send_image(array_data, websocket, mode):
+    max_dimension = 2500 #maximum pixel width or height to be sent out. Increase if higher fidelity is needed
+    try:
+        if array_data.shape[0] > max_dimension or array_data.shape[1] > max_dimension:
```

Comment:
You don't want to be running all of this buffer and image manipulation in an `async` function. I would:

- remove `async` keyword from this function
- pull the `await websocket.send_bytes(buffered.getvalue())` call out of this function
- call this function with `asyncio.run_thread()`